### PR TITLE
fix: setRequestInterception and setExtraHTTPHeaders not working together (#1729)

### LINKED ISSUE #1729: page.goto returns a promise to null when intercepting request and modifying referer together
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 0.13.0 (but also occurs at puppeteer@next )
* Platform / OS version: Mac OS
* URLs (if applicable): *
* Node.js version: 8.9.1

**What steps will reproduce the problem?**

1. It works
```js
const URL = 'http://example.com/';
const REFERER = 'http://example.net/';
(async() => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    // await page.setRequestInterception(true);
    // page.on('request', interceptedRequest => {
    //  interceptedRequest.continue();
    // });
    await page.setExtraHTTPHeaders({ 'referer': REFERER });
    const response = await page.goto(URL);
    try {
      const text = await response.text();
      console.log(text); // goes here
    } catch(error) {
      console.error(error);
    }
    await page.screenshot({path: 'example.png'});
    await page.close();
    await browser.close();
})();
```

2. It works
```js
const URL = 'http://example.com/';
const REFERER = 'http://example.net/';
(async() => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setRequestInterception(true);
    page.on('request', interceptedRequest => {
     interceptedRequest.continue();
    });
    // await page.setExtraHTTPHeaders({ 'referer': REFERER });
    const response = await page.goto(URL);
    try {
      const text = await response.text();
      console.log(text); // goes here
    } catch(error) {
      console.error(error);
    }
    await page.screenshot({path: 'example.png'});
    await page.close();
    await browser.close();
})();
```

3. but it catches the error
```js
const URL = 'http://example.com/';
const REFERER = 'http://example.net/';
(async() => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setRequestInterception(true);
    page.on('request', interceptedRequest => {
     interceptedRequest.continue();
    });
    await page.setExtraHTTPHeaders({ 'referer': REFERER });
    const response = await page.goto(URL);
    try {
      const text = await response.text();
      console.log(text);
    } catch(error) {
      console.error(error); // goes here
    }
    await page.screenshot({path: 'example.png'});
    await page.close();
    await browser.close();
})();
```

The most strange thing is that if I modify other headers than referer, it works.

4. it works
```js
await page.setExtraHTTPHeaders({ 'otherthanreferrer': REFERER });
```

Either way, the screenshots are taken ok, so it's not like failing to navigating to the next page.

**What is the expected result?**

`page.goto()` always returns a promise to response

**What happens instead?**

`page.goto()` returns a promise to `null`

```
TypeError: Cannot read property 'text' of null
```

### PR DESCRIPTION
**Pull request by yujiosaka**
This is just a fix to the problem that `page.goto` resolves to `null`.
Setting referer does not work properly, but that should be discussed [here](https://github.com/GoogleChrome/puppeteer/issues/469).

Fixes: https://github.com/GoogleChrome/puppeteer/issues/1729

### CODE REVIEW

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -569,12 +569,16 @@ function generateRequestHash(request) {
   };
 
   if (!normalizedURL.startsWith('data:')) {
-    const headers = Object.keys(request.headers);
-    headers.sort();
-    for (const header of headers) {
```
Comment (aslushnikov) [inline lib/NetworkManager.js]: can we please minimize the change? I believe you can add a `header.toLowerCase()` as the first line in the loop:
```js
for (let header of headers) {
  header = header.toLowerCase();
  if (key === 'accept' || key === 'referer' ...)
}
```

File: test/test.js [hunk 2]
```diff
@@ -1321,6 +1321,16 @@ describe('Page', function() {
       const response = await page.goto(server.EMPTY_PAGE);
       expect(response.ok()).toBe(true);
     });
+    it('should works with customizing referer headers', async({page, server}) => {
+      await page.setExtraHTTPHeaders({ 'referer': server.EMPTY_PAGE });
```
Comment (aslushnikov) [inline test/test.js:1325]: this is a good test, thank you.

**Review by aslushnikov — COMMENTED** [review]
Thank you @yujiosaka for taking time to work on this.

**Review by aslushnikov — APPROVED** [review]
@yujiosaka I've slightly changed your code; I hope you don't mind. Thank you for fixing this!

**Comment by yujiosaka** [comment]
@aslushnikov thanks!

**Pull request merged**